head.S

```diff
@@ -256,13 +256,22 @@ ENDDATA(l4_identmap)
 
 	.section .bootloader_data, "a", @progbits
 
-GLOBAL(lz_header) /* The LZ header setup by the bootloader */
-	.long	0x8e26f178 /* UUID */
+GLOBAL(lz_header)		/* Set up by the bootloader, keep in sync with boot.h */
+	.long	0x8e26f178	/* UUID */
 	.long	0xe9119204
 	.long	0x5bc82a83
 	.long	0x02ccc476
-	.long	0 /* Total size of Trenchboot Intermediate Loader */
-		  /* bzImage (padded out to next page) */
-	.long	0 /* Zero Page address */
-	.fill	0x14 /* MSB Key Hash */
+	.long	0			/* Total size of Trenchboot Intermediate Loader */
+						/* bzImage (padded out to next page) */
+	.long	0			/* Zero Page address */
+	.long	0			/* Event Log address */
+	.long	0			/* Event Log size */
+	.fill	0x14		/* MSB Key Hash */
+
+	/* Hashes of LZ to be written to DRTM TPM event log */
```

Comment:
Why not have a field for algo id and then reserve space for sha512 (largest hash)?

Reply:
This right now has the exact layout as is used by new (agile) log format, so it can be written with simple `memcpy`, while old (SHA1) format takes just one of these. We need both because we don't know the TPM family in advance.

Reply:
After the discussion on Wire I decided to change it with new bootloader data structure in one of the following PRs, for now I'd rather leave it as it is to have some base to work on